sherpa-onnx/csrc/sherpa-onnx.cc

```diff
@@ -71,63 +62,68 @@ for a list of pre-trained models to download.
 
   sherpa_onnx::OnlineRecognizer recognizer(config);
 
-  int32_t sampling_rate = -1;
+  float duration = 0;
+  for (int32_t i = 1; i <= po.NumArgs(); ++i) {
+    const std::string wav_filename = po.GetArg(i);
+    int32_t sampling_rate = -1;
 
-  bool is_ok = false;
-  std::vector<float> samples =
-      sherpa_onnx::ReadWave(wav_filename, &sampling_rate, &is_ok);
+    bool is_ok = false;
+    const std::vector<float> samples =
+        sherpa_onnx::ReadWave(wav_filename, &sampling_rate, &is_ok);
 
-  if (!is_ok) {
-    fprintf(stderr, "Failed to read %s\n", wav_filename.c_str());
-    return -1;
-  }
-  fprintf(stderr, "sampling rate of input file: %d\n", sampling_rate);
+    if (!is_ok) {
+      fprintf(stderr, "Failed to read %s\n", wav_filename.c_str());
+      return -1;
+    }
+    fprintf(stderr, "sampling rate of input file: %d\n", sampling_rate);
 
-  float duration = samples.size() / static_cast<float>(sampling_rate);
+    const float duration = samples.size() / static_cast<float>(sampling_rate);
 
-  fprintf(stderr, "wav filename: %s\n", wav_filename.c_str());
-  fprintf(stderr, "wav duration (s): %.3f\n", duration);
+    fprintf(stderr, "wav filename: %s\n", wav_filename.c_str());
+    fprintf(stderr, "wav duration (s): %.3f\n", duration);
 
-  auto begin = std::chrono::steady_clock::now();
-  fprintf(stderr, "Started\n");
+    auto begin = std::chrono::steady_clock::now();
+    fprintf(stderr, "Started\n");
 
-  auto s = recognizer.CreateStream();
-  s->AcceptWaveform(sampling_rate, samples.data(), samples.size());
+    auto s = recognizer.CreateStream();
+    s->AcceptWaveform(sampling_rate, samples.data(), samples.size());
 
-  std::vector<float> tail_paddings(static_cast<int>(0.3 * sampling_rate));
-  // Note: We can call AcceptWaveform() multiple times.
-  s->AcceptWaveform(sampling_rate, tail_paddings.data(), tail_paddings.size());
+    std::vector<float> tail_paddings(static_cast<int>(0.3 * sampling_rate));
+    // Note: We can call AcceptWaveform() multiple times.
+    s->AcceptWaveform(
+      sampling_rate, tail_paddings.data(), tail_paddings.size());
 
-  // Call InputFinished() to indicate that no audio samples are available
-  s->InputFinished();
+    // Call InputFinished() to indicate that no audio samples are available
+    s->InputFinished();
 
-  while (recognizer.IsReady(s.get())) {
-    recognizer.DecodeStream(s.get());
-  }
+    while (recognizer.IsReady(s.get())) {
```

Comment:
@csukuangfj I adjusted the `fprintf` statements so that they are not timed. 

Regarding the decoding of multiple files in batches, I can work on that for sure. But that is a bit beyond the original scope of this PR. Can I work on that after the current PR is merged? Thanks.

Reply:
Sure.